components/test/brave_welcome_ui/actions/welcome_actions_test.ts

```diff
@@ -29,4 +29,34 @@ describe('welcome_actions', () => {
       payload: undefined
     })
   })
+
+  it('changeDefaultSearchProvider', () => {
+    expect(actions.changeDefaultSearchProvider(12345)).toEqual({
+      type: types.CHANGE_DEFAULT_SEARCH_PROVIDER,
+      payload: 12345
+    })
+  })
+
+  it('getSearchEngineProvidersStarted', () => {
+    expect(actions.getSearchEngineProvidersStarted()).toEqual({
+      type: types.IMPORT_DEFAULT_SEARCH_PROVIDERS_STARTED
+    })
+  })
+
+  it('getSearchEngineProvidersSuccess', () => {
+    const mockPayload = []
+    expect(actions.getSearchEngineProvidersSuccess(mockPayload)).toEqual({
+      type: types.IMPORT_DEFAULT_SEARCH_PROVIDERS_SUCCESS,
+      payload: mockPayload
+    })
+  })
+
+  it('getSearchEngineProvidersFailure', () => {
+    expect(actions.getSearchEngineProvidersFailure()).toEqual({
+      type: types.IMPORT_DEFAULT_SEARCH_PROVIDERS_FAILURE
+    })
+  })
+
+  // TODO: getSearchEngineProviders can be tested via mocking the entire
+  // store. Would require redux-mock-store as dependency
```

Comment:
@cezaraugusto ☝️thoughts on using https://github.com/dmitry-zaets/redux-mock-store to test the async stuff?

Reply:
👍I've never used it but please add if doable and you think it would help us with testing